dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowAction.java

```diff
@@ -1,155 +1,273 @@
 		return null;
 	}
+
 	@JsonIgnore
 	public String getPermissionType() {
 		return this.getClass().getCanonicalName();
 	}
+
 	@JsonIgnore
 	public boolean isParentPermissionable() {
 		return false;
 	}
-	private static final long serialVersionUID = 1L;
 
-	String id;
-	String name;
-	String stepId;
-	String condition;
-	String nextStep;
-	String nextAssign;
-	String icon;
-	boolean roleHierarchyForAssign;
-	boolean requiresCheckout;
-	
+	/**
+	 * @deprecated see {@link #getRequiresCheckoutOption()}
+	 * @return boolean
+	 */
 	public boolean requiresCheckout() {
 		return requiresCheckout;
 	}
+
+	/**
+	 * @deprecated see {@link #getRequiresCheckoutOption()}
+	 * @return boolean
+	 */
+	@Deprecated
 	public boolean isRequiresCheckout() {
 		return requiresCheckout;
     }
+
+	/**
+	 * @deprecated see {@link #setRequiresCheckoutOption(String)}
+	 * p-pp0()}
+	 * @param requiresCheckout
+	 */
+	@Deprecated
 	public void setRequiresCheckout(boolean requiresCheckout) {
 		this.requiresCheckout = requiresCheckout;
 	}
+
+	/**
+	 * Option for requires checkout:
+	 * WorkflowAction.LOCKED
+	 * WorkflowAction.UNLOCKED
+	 * WorkflowAction.LOCKED_OR_UNLOCKED
+	 * @return String
+	 */
+	public String getRequiresCheckoutOption() {
+		return requiresCheckoutOption;
+	}
+
+	/**
+	 * Set the option for the requires checkout, valid options:
+	 * WorkflowAction.LOCKED
+	 * WorkflowAction.UNLOCKED
+	 * WorkflowAction.LOCKED_OR_UNLOCKED
+	 * @param requiresCheckoutOption
+	 */
+	public void setRequiresCheckoutOption(String requiresCheckoutOption) {
+
+		if (REQUIRES_LOCK_OPTION_SET.contains(requiresCheckoutOption)) {
+			this.requiresCheckoutOption = requiresCheckoutOption;
+		} else {
+			throw new IllegalArgumentException(
+					"Invalid option for RequiresCheckoutOption, valid options: " + REQUIRES_LOCK_OPTION_SET);
+		}
+	}
+
 	public boolean isRoleHierarchyForAssign() {
 		return roleHierarchyForAssign;
 	}
+
 	public void setRoleHierarchyForAssign(boolean roleHierarchyForAssign) {
 		this.roleHierarchyForAssign = roleHierarchyForAssign;
 	}
+
 	public String getIcon() {
 		if(!UtilMethods.isSet(icon)){
 			return "workflowIcon";
 		}
 		return icon;
 	}
+
 	public void setIcon(String icon) {
 		this.icon = icon;
 	}
-	int order;
-	boolean assignable;
-	boolean commentable;
+
 	public String getId() {
 		return id;
 	}
+
 	public void setId(String id) {
 		this.id = id;
 	}
+
 	public String getName() {
 		return name;
 	}
+
 	public void setName(String name) {
 		this.name = name;
 	}
+
+	/**
+	 * Reflects the old relation between Action and step
+	 * @deprecated this is keep just by legacy reason, new apps should not use it
+	 */
+	@Deprecated
 	public String getStepId() {
 		return stepId;
 	}
+
+	/**
+	 * Reflects the old relation between Action and step
+	 * @deprecated this is keep just by legacy reason, new apps should not use it
+	 */
+	@Deprecated
 	public void setStepId(String stepId) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)